src/setContextLink.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  Operation,
+  NextLink,
+  FetchResult,
+} from './types';
+import * as Observable from 'zen-observable';
+
+import {
+  ApolloLink,
+} from './link';
+
+export default class SetContextLink extends ApolloLink {
+
+  constructor (private context?: any) {
+    super();
+  }
+
+  public request(operation: Operation, forward: NextLink): Observable<FetchResult> {
+    if (!operation.context) {
+      operation.context = {};
+    }
+    operation.context = {
```

Comment:
fwiw, recursive merges are really expensive :( I would say that this link overwrites and previous context of the same keys when being added.

@evanshauser when we call request within the `ApolloLink` interface, thoughts on creating a copy of the operation and passing it down?